src/access/wac.ts

```diff
@@ -292,6 +297,24 @@ async function setActorAccess<T extends WithServerResourceInfo>(
   readAccess: typeof getAgentAccessWac,
   writeAccess: typeof setAgentResourceAccessWac,
   options: Partial<typeof internal_defaultFetchOptions>
+): Promise<(T & WithResourceAcl) | null>;
+async function setActorAccess<T extends WithServerResourceInfo>(
+  resource: T,
+  actor: undefined,
+  access: WacAccess,
+  readAccess: typeof getPublicAccessWac,
+  writeAccess: typeof setPublicResourceAccessWac,
+  options: Partial<typeof internal_defaultFetchOptions>
+): Promise<(T & WithResourceAcl) | null>;
```

Comment:
I don't think the overloads actually add anything here? (Though might be missing something?)

Reply:
It's intended to ensure that either the function is called with an `undefined` actor, and with a `readAccess` function that matches `typeof getPublicAccessWac`, or with a proper actor and a function that matches `typeof getActorAccessWac`, which expects a non-undefined actor. However, I think you're right, it looks like it's getting to a point where that's not the right abstraction, and that having this function be unique instead of two functions, one for actors and one for actor classes would make more sense.

Reply:
`setActorAccess` is now split in two functions, one for actors and one for actor classes, with the common code factored in its own functions. See c685ad7893dec1789efbe2d4b5f8e2858ae723d2

Reply:
Ah of course, thanks for the explanation still :)